assets/references.go

```diff
@@ -11,12 +12,23 @@ func init() {
 	utils.Validator.RegisterStructValidation(LabelReferenceValidation, LabelReference{})
 }
 
+// Reference is interface for all reference types
+type Reference interface {
```

Comment:
this feels weird but maybe that's just the Java programmer in me. I have no way to explicitly make all the asset reference types be a particular interface.. anything that implements `String()` could pretend to be `asset.Reference`.. but what are the chances of that happening?

Reply:
String() seems a bit broad for sure? Do all references have UUID()'s? Maybe they all have a Type()?

Reply:
Unfortunately not (e.g. `FieldReference` has `key`)... they don't have any methods and if I start adding methods then they'll collide with the field names and I'll have to add envelopes for deserialization etc... argh.

Maybe the best way to think about this.. is what are we going to do with these? What functionality does the caller of `ReadSession` want on them. If it's just going to stringify them and log 'em, then maybe it's fine that it's essentially getting a bunch of Stringers? Honestly not sure what's the most go-ey approach here...

Reply:
If References represent identity of an object, then it seems we would want the interface to be related to that somehow. Stringer says anything with that method matches which doesn't feel right. Maybe it is `Identity()` that returns a `string` that is the id in some way? (ie either UUID or label or key?) That feels less weird to me and gives us some hope of still having the compiler catching us doing stupid things.

Reply:
Added `Type()` and `Identity()`